presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -687,8 +687,8 @@ public int getRunningTaskCount()
     {
         if (subGroups().isEmpty()) {
             return runningQueries.stream()
-                    .filter(SqlQueryExecution.class::isInstance)
-                    .mapToInt(query -> ((SqlQueryExecution) query).getRunningTaskCount())
+                    .filter(LocalDispatchQuery.class::isInstance)
```

Comment:
Just so I understand, the core of this bug is that `SqlQueryExecution.getRunningTaskCount()` always returns 0?

Reply:
The core reason is, query instances in InternalResourceGroup.runningQueries is of type LocalDispatchQuery, not of type SqlQueryExecution, so the filter will filter out all running query instances.

Reply:
Oh, good catch!

Reply:
Let's add `getRunningTaskCount` to `ManagedQueryExecution` interface and check instances of the interface.